docs: KB refresh from Zendesk tickets (Jun 1 – Jul 31, 2026) (#2995)
wandb/docs: support/models/articles/how-do-i-delete-many-artifacts-at-once.mdx

---
title: "How do I delete many artifacts at once?"

> keywords: ["Artifacts", "Storage", "API"]

Reviewer: "API" is too generic. Could refer to SCIM API.

```suggestion
keywords: ["Artifacts", "Storage", "Python API"]
```

---

When you need to remove a large number of artifacts, for example to get back under your storage limit, deleting them in the W&B App becomes impractical. Large deletions can time out with a `context deadline exceeded` error, and the artifact list isn't built for clearing hundreds of versions.

Use the [W&B Public API](/models/ref/python/public-api/api) instead. The following script deletes every version of one artifact type in a project.

> Replace the `[YOUR-TEAM]` and `[YOUR-PROJECT]` placeholders with your own values. Use your team or username for `[YOUR-TEAM]`, not your organization name. Passing an organization name fails with `ValueError: Unable to parse 'ArtifactTypes' response data`.

Reviewer: Not sure why this was added:

```suggestion
Replace the `[YOUR-TEAM]` and `[YOUR-PROJECT]` placeholders with your own values. Use your team or username for `[YOUR-TEAM]`, not your organization name.
```

Author: I believe it was added to correlate to a ticket - e.g. more than one customer i think was trying to pass organization instead of team, and encountered that error. But you're right, it is superfluous.

<Warning>

> Artifact deletion is irreversible. The script passes `delete_aliases=True`, so it also deletes versions that carry an alias such as `latest`. Keep `DRY_RUN = True` and review the output before you set it to `False`.

Reviewer: ```suggestion
Artifact deletion is destructive. When deleting source artifact versions, W&B deletes the artifact and its files. Passing `delete_aliases=True` enables deletion of versions with aliases such as `latest`, and removes those aliases. 

Use `DRY_RUN = True` first, review the printed artifact names, and only then set DRY_RUN = False.
```

</Warning>

```python
import wandb

# Assumes you've run `wandb login` already.

> entity = "[YOUR-TEAM]"  # your username or team

Reviewer: Let's agree on placeholder variables? https://github.com/coreweave/docs-skills/pull/62#discussion_r3691433460

Author: oh, i thought we had. to these.
Noted. I guess it's an ongoing TBD discussion.

project = "[YOUR-PROJECT]"

> artifact_type = "model"

Reviewer: This assumes artifacts of type model. Should explain what this means, that you should also replace (in addition to entity and project) the type of artifact to delete.

Author: excellent catch, thanks. i meant to call this out, and then it slipped past me...

# Set to False to actually delete. Deletion is irreversible.
DRY_RUN = True

api = wandb.Api()

for collection in api.artifact_collections(f"{entity}/{project}", artifact_type):
    for version in collection.artifacts():
        if DRY_RUN:
            print(f"[dry-run] would delete {version.name}")
        else:
            print(f"Deleting {version.name}")
            version.delete(delete_aliases=True)
```

The script deletes versions, not collections. An emptied collection still appears in the project's **Artifacts** tab.

> `api.artifact_collections()` returns link collections as well as source collections. When a version comes from a link collection, `version.delete(delete_aliases=True)` removes only the link. The source artifact and its files stay where they are, so that version frees no storage.

Reviewer: Incorrect

```suggestion
`wandb.Api.artifact_collections()` returns link collections as well as source collections. When a version comes from a link collection, `version.delete(delete_aliases=True)` removes only the link. The source artifact and its files stay where they are, so that version frees no storage.
```

> `api.artifact_collections()` returns link collections as well as source collections. When a version comes from a link collection, `version.delete(delete_aliases=True)` removes only the link. The source artifact and its files stay where they are, so that version frees no storage.

Reviewer: Should probably elaborate what we mean by "link collections" vs "source collections".

## Reclaim the space

> Deleted artifacts are soft-deleted first, then removed during garbage collection, so your storage meter doesn't drop immediately. For the timing and how to verify your usage, see [Why does the storage meter not update after deleting runs?](/support/models/articles/why-does-the-storage-meter-not-update-af) and [Delete an artifact](/models/artifacts/delete-artifacts).

Reviewer: Dumb question: @mdlinville do we refer to storage as "storage meter"?

Author: valid question - tho links to an existing article, so i didn't question that. if we don't, we should update that one too.

Reviewer: I'll defer to Matt w.r.t. to whether or not they ("storage meter") should get updated in this article and the other.

Reviewer 2: I don't think we call it that but I see where the agent is coming from. It's responding to usage metering, which visualizes your usage in the organization dashboard. I'd probably call this your storage usage or something like that, whatever the relevant docs say. This brings up that the agent can take cues from the pages it's pulling from. It's not going to blow out the context and will probably lead to less creativity in coining new terms.

---
title: "How do I delete many artifacts at once?"
keywords: ["Artifacts", "Storage", "Python API"]
---

When you need to remove a large number of artifacts, for example to get back under your storage limit, deleting them in the W&B App becomes impractical. Large deletions can time out with a `context deadline exceeded` error, and the artifact list isn't built for clearing hundreds of versions.

Use the [W&B Public API](/models/ref/python/public-api/api) instead. The following script deletes every version of one artifact type in a project.

Replace the `[YOUR-TEAM]` and `[YOUR-PROJECT]` placeholders with your own values. Use your team or username for `[YOUR-TEAM]`, not your organization name.

<Warning>
Artifact deletion is destructive. When deleting source artifact versions, W&B deletes the artifact and its files. Passing `delete_aliases=True` enables deletion of versions with aliases such as `latest`, and removes those aliases. 

Use `DRY_RUN = True` first, review the printed artifact names, and only then set DRY_RUN = False.
</Warning>

```python
import wandb

# Assumes you've run `wandb login` already.

> entity = "[YOUR-TEAM]"  # your username or team

Author: ```suggestion
entity = "[YOUR-TEAM]"  # Replace with your team.
```

> project = "[YOUR-PROJECT]"

Author: ```suggestion
project = "[YOUR-PROJECT]"  # Replace with your project.
```

> artifact_type = "model"

Author: ```suggestion
artifact_type = "model"  # This deletes artifacts of type 'model'. Replace with the type of artifact you want to delete. 
```

---
title: "How do I delete many artifacts at once?"
keywords: ["Artifacts", "Storage", "Python API"]
---

When you need to remove a large number of artifacts, for example to get back under your storage limit, deleting them in the W&B App becomes impractical. Large deletions can time out with a `context deadline exceeded` error, and the artifact list isn't built for clearing hundreds of versions.

> Use the [W&B Public API](/models/ref/python/public-api/api) instead. The following script deletes every version of one artifact type in a project.

Reviewer: ```suggestion
When you need to delete many artifact versions, such as to reduce storage usage, use the [W&B Public API](/models/ref/python/public-api/api) instead of the W&B App. Bulk deletions in the App can time out with a `context deadline exceeded` error, and the artifact list is not designed for deleting hundreds of versions.

The following script deletes all artifact versions of a specified type in a project.
```

Replace the `[YOUR-TEAM]` and `[YOUR-PROJECT]` placeholders with your own values. Use your team or username for `[YOUR-TEAM]`, not your organization name.

<Warning>
Artifact deletion is destructive. When deleting source artifact versions, W&B deletes the artifact and its files. Passing `delete_aliases=True` enables deletion of versions with aliases such as `latest`, and removes those aliases. 

Use `DRY_RUN = True` first, review the printed artifact names, and only then set DRY_RUN = False.
</Warning>

```python
import wandb

# Assumes you've run `wandb login` already.

> entity = "[YOUR-TEAM]"  # Replace with your team.

Reviewer: TBD what the verdict s here. To unblock, we can use the current W&B doc conventions.

Author: okay, yes that was my plan.  This syntax is what we had agreed upon as a team (it is already in the style guide skill).  I mentioned to John that you wanted to reopen this discussion, so we'll probably talk about it in the next Internal Skills WG meeting.  perhaps you'd like to attend as a guest to aide in this discussion?)
Regardless, yes, for this PR I think using this is appropriate for now - if we change our agreed-upon convention, a LOT of places will need to be updated, beyond just this.

Reviewer: Yup! Add me to the meeting

project = "[YOUR-PROJECT]"  # Replace with your project.

> artifact_type = "model"  # This deletes artifacts of type 'model'. Replace with the type of artifact you want to delete. 

Reviewer: ```suggestion
artifact_type = "model"  # Replace with the artifact type to delete.
```

> # Set to False to actually delete. Deletion is irreversible.

Reviewer: ```suggestion
# Keep this set to True until you review the versions to delete.
```

DRY_RUN = True

api = wandb.Api()

for collection in api.artifact_collections(f"{entity}/{project}", artifact_type):
    for version in collection.artifacts():
        if DRY_RUN:
            print(f"[dry-run] would delete {version.name}")
        else:
            print(f"Deleting {version.name}")
            version.delete(delete_aliases=True)
```

> The script deletes versions, not collections. An emptied collection still appears in the project's **Artifacts** tab.

Reviewer: ```suggestion
The script deletes artifact versions, not collections. An emptied collection still appears in the project's **Artifacts** tab.
```

> `wandb.Api.artifact_collections()` returns link collections as well as source collections. A source collection holds the artifact versions logged to a run in your project. A link collection holds pointers to those versions, published to a registry so the rest of your organization can use them. See [Link an artifact version to a collection](/models/registry/link_version).

Reviewer: ```suggestion
`wandb.Api.artifact_collections()` returns both source collections and link collections:

- A source collection contains artifact versions logged to runs in the project.
- A link collection contains references to artifact versions linked to another collection, such as a collection in a registry.

For more information, see [Link an artifact version to a collection](/models/registry/link_version).
```
